kubectl-pg/cmd/scale.go

```diff
@@ -0,0 +1,151 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+	minInstances, maxInstances := allowedMinMaxInstances(config)
+	if minInstances == -1 && maxInstances == -1 {
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else if numberOfInstances <= maxInstances && numberOfInstances>= minInstances {
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else if minInstances == -1 && numberOfInstances < postgresql.Spec.NumberOfInstances ||
+		 maxInstances == -1 && numberOfInstances > postgresql.Spec.NumberOfInstances{
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else {
+		log.Fatalf("cannot scale to the provided instances as they don't adhere to MIN_INSTANCES: %v and MAX_INSTANCES: %v provided in configmap or operatorconfiguration",maxInstances,minInstances)
+	}
+	if numberOfInstances == 0 {
+		fmt.Printf("Scaling to zero leads to down time. please type %s/%s and hit Enter this serves to confirm the action\n", namespace, clusterName)
+		confirmAction(clusterName, namespace)
+	}
+	UpdatedPostgres, err := postgresConfig.Postgresqls(namespace).Update(postgresql)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if UpdatedPostgres.ResourceVersion != postgresql.ResourceVersion {
+		fmt.Printf("scaled postgresql %s/%s to %d instances\n", UpdatedPostgres.Namespace, UpdatedPostgres.Name, UpdatedPostgres.Spec.NumberOfInstances)
+		return
+	}
+	fmt.Printf("postgresql %s is unchanged.\n", postgresql.Name)
+}
+
+func allowedMinMaxInstances(config *rest.Config) (int32, int32){
+	k8sClient,err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	operator,err :=k8sClient.AppsV1().Deployments(getCurrentNamespace()).Get("postgres-operator",metav1.GetOptions{})
+	if err!= nil {
+		log.Fatal(err)
+	}
+	operatorContainer:=operator.Spec.Template.Spec.Containers
+	var configMapName, operatorConfigName string
+	minInstances := -1
+	maxInstances := -1
+	for _,envData := range operatorContainer[0].Env {
+		if envData.Name == "CONFIG_MAP_NAME"  {
+			configMapName = envData.Value
+		}
+		if envData.Name == "POSTGRES_OPERATOR_CONFIGURATION_OBJECT" {
+			operatorConfigName = envData.Value
+		}
+	}
+	if operatorConfigName == "" {
+		configMap, err := k8sClient.CoreV1().ConfigMaps(getCurrentNamespace()).Get(configMapName, metav1.GetOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+		configMapData := configMap.Data
+		for key, value := range configMapData {
+			if key == "min_instances" {
+				minInstances,err = strconv.Atoi(value)
+				if err != nil {
+					log.Fatalf("invalid min instances in configmap %v",err)
+				}
+			}
+			if key == "max_instances" {
+				maxInstances,err = strconv.Atoi(value)
+				if err != nil {
+					log.Fatalf("invalid max instances in configmap %v",err)
+				}
+			}
+		}
+	} else if configMapName == "" {
+		pgClient,err :=	PostgresqlLister.NewForConfig(config)
+		if err != nil {
+			log.Fatal(err)
+		}
+		operatorConfig,err := pgClient.OperatorConfigurations("s").Get(operatorConfigName,metav1.GetOptions{})
```

Comment:
`OperatorConfigurations("s")` should be called with the `namespace` parameter, and not some hard-coded value.

Reply:
some typo, changed it ```getCurrentNamespace()```